### PR TITLE
Fix rbac for openshift machines

### PR DESCRIPTION
**Pull request by alvaroaleman**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**Documentation**:

**Release note**:

```release-note
none
```

### CODE REVIEW

File: api/pkg/resources/resources.go [hunk 1]
```diff
@@ -358,11 +358,11 @@ type ECDSAKeyPair struct {
 	Cert *x509.Certificate
 }
 
-// RoleCreator defines an interface to create/update RBAC Roles
-type RoleCreator = func(data RoleDataProvider, existing *rbacv1.Role) (*rbacv1.Role, error)
+// RoleCreatorDeprecated defines an interface to create/update RBAC Roles
+type RoleCreatorDeprecated = func(data RoleDataProvider, existing *rbacv1.Role) (*rbacv1.Role, error)
 
 // RoleBindingCreator defines an interface to create/update RBAC RoleBinding's
```
Comment (xrstf) [inline api/pkg/resources/resources.go]: You've missed updating the comment for this function.

**Comment by alvaroaleman** [comment]
@xrstf Updated, PTAL

**Pull request merged**